src/Compilers/CSharp/Portable/Utilities/IValueSet.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// An interface representing a set of values of a specific type.  During construction of the state machine
+    /// for pattern-matching, we track the set of values of each intermediate result that can reach each state.
+    /// That permits us to determine when tests can be eliminated either because they are impossible (can be
+    /// replaced by an always-false test) or always true with the set of values that can reach that state (and
+    /// can be replaced by an always true test).
+    /// </summary>
+    internal interface IValueSet
+    {
+        /// <summary>
+        /// Return the intersection of this value set with another. Both must have been created with the same <see cref="IValueSetFactory{T}"/>.
+        /// </summary>
+        IValueSet Intersect(IValueSet other);
+
+        /// <summary>
+        /// Return this union of this value set with another. Both must have been created with the same <see cref="IValueSetFactory{T}"/>.
+        /// </summary>
+        IValueSet Union(IValueSet other);
+
+        /// <summary>
+        /// Return the complement of this value set.
+        /// </summary>
+        IValueSet Complement();
+
+        /// <summary>
+        /// Test if the value set contains any values that satisfy the given relation with the given value.  Supported values for <paramref name="relation"/>
+        /// Are <see cref="BinaryOperatorKind.Equal"/> for all supported types, and for numeric types (except decimal) we also support
```

Comment:
>Are [](start = 12, length = 3)

Lowercase "are"

Reply:
Will address in the next PR.

---
In reply to: [383044194](https://github.com/dotnet/roslyn/pull/41503#discussion_r383044194) [](ancestors = 383044194)